drivers/include/onewire.h

```diff
@@ -0,0 +1,445 @@
+ * @param[in] rom       1-Wire ROM code of target device
+ *
+ * @retval  0 on success
+ * @retval  -ENXIO if no slave answered the reset sequence
+ */
+int onewire_select(onewire_t *bus, const onewire_rom_t *rom);
+
+/**
+ * @brief   Read data from the bus
+ *
+ * @param[in] bus       1-Wire bus device descriptor
+ * @param[out] data     buffer to write received bytes into
+ * @param[in] len       number of bytes to read from the bus
+ *
+ * @retval  0 on success
+ * @retval  -EIO on failure
+ */
+int onewire_read(onewire_t *bus, void *data, size_t len);
+
+/**
+ * @brief   Convenience function for reading a single byte from the bus
+ *
+ * @param[in] bus       1-Wire bus device descriptor
+ * @param[out] data     byte read from the bus
+ *
+ * @retval  0 on success
+ * @retval  -EIO on failure
+ */
+static inline int onewire_read_byte(onewire_t *bus, uint8_t *data)
+{
+    return onewire_read(bus, data, 1);
+}
+
+/**
+ * @brief   Convenience function for reading a 16 bit word from the bus
+ *
+ * @param[in] bus       1-Wire bus device descriptor
+ * @param[out] data     word read from the bus
+ *
+ * @retval  0 on success
+ * @retval  -EIO on failure
+ */
+static inline int onewire_read_word(onewire_t *bus, uint16_t *data)
+{
+    le_uint16_t buffer;
+
+    const int res = onewire_read(bus, &buffer, 2);
+    *data = byteorder_ltohs(buffer);
+    return res;
+}
+
+/**
+ * @brief   Write data to the bus
+ *
+ * @param[in] bus       1-Wire bus device descriptor
+ * @param[in] data      buffer holding the data that is written to the bus
+ * @param[in] len       number of bytes to write to the bus
+ *
+ * @retval  0 on success
+ * @retval  -EIO on failure
+ */
+int onewire_write(onewire_t *bus, const void *data, size_t len);
+
+/**
+ * @brief   Convenience function for writing a single byte to the bus
+ *
+ * @param[in] bus       1-Wire bus device descriptor
+ * @param[in] data      byte to write to the bus
+ *
+ * @retval  0 on success
+ * @retval  -EIO on failure
+ */
+static inline int onewire_write_byte(onewire_t *bus, uint8_t data)
+{
+    return onewire_write(bus, &data, 1);
+}
+
+/**
+ * @brief   Convenience function for writing a 16 bit word to the bus
+ *
+ * @param[in] bus       1-Wire bus device descriptor
+ * @param[in] data      word to write to the bus
+ *
+ * @retval  0 on success
+ * @retval  -EIO on failure
+ */
+static inline int onewire_write_word(onewire_t *bus, uint16_t data)
+{
+    const le_uint16_t buffer = byteorder_htols(data);
+
+    return onewire_write(bus, &buffer, 2);
+}
+
+/**
+ * @brief   Calculate a 8 bit CRC using the 1-Wire specific polynomial
+ *
+ * @param[in] seed      the seed (starting value) for the checksum
+ * @param[in] data      input data
+ * @param[in] len       len of @p data in bytes
+ *
+ * @return  calculated 8 bit CRC
+ */
+static inline uint_fast8_t onewire_crc8(uint_fast16_t seed, const void *data,
+    size_t len)
+{
+    return crc8(data, len, 0x80, seed);
```

Comment:
Hmm, I think I am not calculating the CRC properly here. Still working on figuring out the right polynomial and how to handle reflecting the input.

Reply:
One thing up front: the polynomial is wrong. CRC8 for 1-Wire uses 0x8C and not 0x80. Actually 0x80 won't generate good CRC checksums anyways.

<hr/>

So it depends on if you *really* want to use the built-in CRC8 function or not.

For the built-in function to work, you have to reflect all the input bytes, the polynomial and the output byte. That is a lot of computation.

I wrote a smöl test program to demonstrate:
```c
#include <checksum/crc8.h>
#include <stdio.h>

// from: https://github.com/nickandrew/avr-onewire/blob/master/maxim-crc8.c
uint8_t crc8_update(uint8_t crc, uint8_t data) {
        uint8_t i;

        crc = crc ^ data;

        for (i = 0; i < 8; ++i) {
                if (crc & 0x01) {
                        crc = (crc >> 1) ^ 0x8c;
                } else {
                        crc >>= 1;
                }
        }
        return crc;
}

uint8_t reflect_byte(uint8_t b) {
    b = (b & 0xF0) >> 4 | (b & 0x0F) << 4;
    b = (b & 0xCC) >> 2 | (b & 0x33) << 2;
    b = (b & 0xAA) >> 1 | (b & 0x55) << 1;
    return b;
}

int main(void) {
    for(uint32_t i = 0; i < 256; i++) {
        uint8_t i_refl = reflect_byte(i);
        uint8_t riot_CRC8 = reflect_byte(crc8((uint8_t*) &i_refl, 1, 0x31, 0x00));

        uint8_t onewire_CRC8 = crc8_update(0, (uint8_t) i);
        uint8_t xor_CRC8 = riot_CRC8 ^ onewire_CRC8;
        printf("i: %03d, CRC8: %02X, Onewire CRC8: %02X, XOR: %02X\n",i, riot_CRC8, onewire_CRC8, xor_CRC8);
    }

    return 0;
}
```

And the output is the following:
```
2025-06-06 15:34:59,666 # RIOT native hardware initialization complete.
2025-06-06 15:34:59,667 #
2025-06-06 15:34:59,668 # main(): This is RIOT! (Version: 2025.04-devel-500-g42e08-add-driver-onewire)
2025-06-06 15:34:59,668 # i: 000, CRC8: 00, Onewire CRC8: 00, XOR: 00
2025-06-06 15:34:59,669 # i: 001, CRC8: 5E, Onewire CRC8: 5E, XOR: 00
2025-06-06 15:34:59,669 # i: 002, CRC8: BC, Onewire CRC8: BC, XOR: 00
2025-06-06 15:34:59,669 # i: 003, CRC8: E2, Onewire CRC8: E2, XOR: 00
2025-06-06 15:34:59,670 # i: 004, CRC8: 61, Onewire CRC8: 61, XOR: 00
2025-06-06 15:34:59,671 # i: 005, CRC8: 3F, Onewire CRC8: 3F, XOR: 00
2025-06-06 15:34:59,671 # i: 006, CRC8: DD, Onewire CRC8: DD, XOR: 00
2025-06-06 15:34:59,671 # i: 007, CRC8: 83, Onewire CRC8: 83, XOR: 00
2025-06-06 15:34:59,672 # i: 008, CRC8: C2, Onewire CRC8: C2, XOR: 00
2025-06-06 15:34:59,672 # i: 009, CRC8: 9C, Onewire CRC8: 9C, XOR: 00
2025-06-06 15:34:59,673 # i: 010, CRC8: 7E, Onewire CRC8: 7E, XOR: 00
2025-06-06 15:34:59,673 # i: 011, CRC8: 20, Onewire CRC8: 20, XOR: 00
2025-06-06 15:34:59,674 # i: 012, CRC8: A3, Onewire CRC8: A3, XOR: 00
2025-06-06 15:34:59,675 # i: 013, CRC8: FD, Onewire CRC8: FD, XOR: 00
2025-06-06 15:34:59,675 # i: 014, CRC8: 1F, Onewire CRC8: 1F, XOR: 00
2025-06-06 15:34:59,676 # i: 015, CRC8: 41, Onewire CRC8: 41, XOR: 00
2025-06-06 15:34:59,676 # i: 016, CRC8: 9D, Onewire CRC8: 9D, XOR: 00
2025-06-06 15:34:59,677 # i: 017, CRC8: C3, Onewire CRC8: C3, XOR: 00
2025-06-06 15:34:59,678 # i: 018, CRC8: 21, Onewire CRC8: 21, XOR: 00
2025-06-06 15:34:59,679 # i: 019, CRC8: 7F, Onewire CRC8: 7F, XOR: 00
2025-06-06 15:34:59,679 # i: 020, CRC8: FC, Onewire CRC8: FC, XOR: 00
2025-06-06 15:34:59,680 # i: 021, CRC8: A2, Onewire CRC8: A2, XOR: 00
...
```

Reply:
```suggestion
    return crc8_lsb(data, len, 0x8C, seed);
```

Once #21543 is merged, you can change it to this. Remember to also change the CRC polynomial.